functions/src/modules/SQR.functions.ts

```diff
@@ -112,124 +112,140 @@ export const sendEmailOnNewAllotment = functions.database
+    completed: original.completed, // timestamp of the submission completion.
+    changed: original.changed, //timestamp of the submission update.
+    author: {
+      name: original.name,
+      emailAddress: original.email_address,
+    },
+  };
+
+  db.ref(`/sqr/submissions/${original.serial}`).update(submission);
+
+  return 1;
+});
+
+
+/**
+ * SQR Process Submission
+ * 1. Notifying the coordinator using a mail that holds the following information
+ * 2. Setting the audio file status
+ * 
+ * Function -> processSubmission()
+ */
+export const processSubmission = functions.database
+.ref('/sqr/submissions/{submission_id}')
+.onCreate(async (snapshot, context) => {
+  const submission = snapshot.val();
+
+  let audioFileStatus = 'WIP';
+  if (submission.cancellation.notPreferredLanguage) 
+    audioFileStatus = 'spare';	 
+  else if (submission.cancellation.audioProblem)	
+    audioFileStatus = 'audioProblem';
 
-export const processSubmissions = functions.database
-  .ref('/webforms/sqr/{submission_id}')
-  .onCreate(async (snapshot, context) => {
-    const original = snapshot.val();
-
-    let audioFileStatus = 'WIP';
-
-    if (original.not_preferred_language) audioFileStatus = 'Spare';
-    else if (original.unable_to_play_or_download)
-      audioFileStatus = 'Audio Problem';
-
-    // 1. Add the webform data to a SQR submissions DB path
-    let submission = {
-      fileName: original.audio_file_name,
-      cancellation: {
-        notPreferredLanguage: audioFileStatus === 'Spare',
-        audioProblem: audioFileStatus === 'Audio Problem',
-      },
-      soundQualityRating: original.sound_quality_rating,
-      unwantedParts: original.unwanted_parts,
-      soundIssues: original.sound_issues,
-      duration: {
-        beginning: original.beginning,
-        ending: original.ending,
-      },
-      comments: original.comments,
-      token: original.token,
-      created: original.created,
-      //  timestamp of the submission creation,
-      // can differ from COMPLETED in case of saving a DRAFT and completing later.
-      completed: original.completed, // timestamp of the submission completion.
-      changed: original.changed, //timestamp of the submission update.
-      devotee: {
-        name: original.name,
-        emailAddress: original.email_address,
-      },
-    };
-    db.ref(`/sqr/submissions/${original.serial}`).update(submission);
-
-    // 2. Update the allotment ( first get the previous NOTES )
-    const filesSnapshot = await db
-      .ref(`/sqr/files/${original.list}/${original.audio_file_name}`)
-      .once('value');
-    let allotmentUpdates = { status: audioFileStatus };
+    //  EXTRACTING the list name first from the file_name
+  let list = helpers.extractListFromFilename(submission.fileName);
+  
+
+  // 2. Update the allotment ( first get the previous NOTES )
+  const filesSnapshot = await db
+    .ref(`/sqr/files/${list}/${submission.fileName}`)
```

Comment:
Just noticed that this path is not correct.
```suggestion
    .ref(`/files/${list}/${submission.fileName}`)
```